sdk/src/azure/core/az_http_response.c

```diff
@@ -186,8 +171,25 @@ az_http_response_get_next_header(az_http_response* ref_response, az_pair* out_he
     //         "_" / "`" / "|" / "~" / DIGIT / ALPHA;
     // any VCHAR,
     //    except delimiters
-    AZ_RETURN_IF_FAILED(
-        _az_span_scan_until(*reader, _az_valid_header_name_to_colon, &field_name_length));
+
+    int32_t input_size = az_span_size(*reader);
+    for (; field_name_length < input_size; ++field_name_length)
+    {
+      uint8_t next_byte = az_span_ptr(*reader)[field_name_length];
+      if (next_byte == ':')
+      {
+        break;
+      }
+      if (!az_http_valid_token[next_byte])
+      {
+        return AZ_ERROR_HTTP_CORRUPT_RESPONSE_HEADER;
+      }
+    }
+
+    if (field_name_length == input_size)
+    {
+      return AZ_ERROR_ITEM_NOT_FOUND;
+    }
```

Comment:
So, making the predicate function unable to handle an error makes almost useless to have the scan function.
It would be better to just removed the `_az_span_scan_until()` function and insert the loop and predicate function directly into each .c file calling `_az_span_scan_until()`

Reply:
We would be removing the function type definition for the predicate as well.. So, less code, I guess.

Reply:
Yes, imo, that would be cleaner and possibly result in smaller code size (or alternatively we change the logic to be different).

Reply:
Fixed in https://github.com/Azure/azure-sdk-for-c/pull/1041